python/plugins/processing/gui/AlgorithmDialog.py

```diff
@@ -98,14 +99,24 @@ def getParameterValues(self):
             if param.flags() & QgsProcessingParameterDefinition.FlagHidden:
                 continue
             if not param.isDestination():
-                wrapper = self.mainWidget().wrappers[param.name()]
-                value = None
-                if wrapper.widget is not None:
-                    value = wrapper.value()
-                    parameters[param.name()] = value
+                try:
+                    wrapper = self.mainWidget().wrappers[param.name()]
+                except KeyError:
+                    continue
+
+                if issubclass(wrapper.__class__, WidgetWrapper):
+                    widget = wrapper.widget
+                else:
+                    widget = wrapper.wrappedWidget()
```

Comment:
Can you add a comment about the supposed class/subclass of `wrapper` in this case?